core/o11y/src/io_tracer.rs

```diff
@@ -187,42 +187,45 @@ impl IoTraceLayer {
         event: &tracing::Event,
         ctx: tracing_subscriber::layer::Context<S>,
     ) {
+        struct FormatSize(Option<u64>);
```

Comment:
cc @jakmeier for an additional pair of eyes here. Though this code is not as perf-sensitive I believe as it is only used inside parameter estimation, so maybe just not making any changes might be better?

Reply:
Yeah IO trace collection is specifically not performance sensitive, as it is mostly just an offline debug tool that is allowed to collect more data than we feasibly could on a live node.

I'm neutral on this one. Optimizing wouldn't hurt but the code feels a bit over-engineered / harder to read. I'd say we either keep the old code or add a few comments at least explaining the new indirection, similar to the comments that already exist on `FormatMillis` and `BlockInfo`.

Reply:
Removed the over-engineered parts.